src/mongo/db/catalog/index_key_validate.cpp

```diff
@@ -269,6 +271,12 @@ StatusWith<BSONObj> validateIndexSpec(
 
     boost::optional<IndexVersion> resolvedIndexVersion;
 
+    // Allow compound hashed index only if FCV is 4.4.
+    const auto isFeatureDisabled =
+                (featureCompatibility.isVersionInitialized() &&
```

Comment:
This call to `isVersionInitialized` also seems off to me. If this code can be called prior to FCV initialization, I would expect the `isFeatureDisabled` boolean to be `true`. If the code cannot be called prior to FCV initialization, leaving it as an embedded invariant inside `getVersion` seems best.

If I'm mistaken on both fronts, a comment describing the scenario would be beneficial.

Reply:
That being said, I now see this is moved from below. @jameswahlin can you verify that logic is actually intended?

Reply:
I believe this is the intended logic. @dgottlieb - your right, this is a bug in the current code. I have informed the author of compound hashed indexes to take a look.

@zhihuiFan, can you replace this with the following? This will correctly disallow hidden indexes if FCV has not been initialized:
const auto isFeatureDisabled =
                (!featureCompatibility.isVersionInitialized() ||
                 featureCompatibility.getVersion() <
                     ServerGlobalParams::FeatureCompatibility::Version::kFullyUpgradedTo44);

Reply:
Hi @jameswahlin :    I'm agreed with this change. Can I handle over this to MongoDB team to change it during the merge.  do we have any other comments?

Reply:
Hi @zhihuiFan - Dan Gottlieb posted a few additional comments as part of his review on March 9th. This was prior to my comment regarding FCV. Would you like MongoDB to address?

Reply:
Yes, Thanks @jameswahlin

Reply:
We will preserve your latest commit and will complete work in additional commits so that you get credit for your work. I will leave this pull request open until we commit your changes so that we can document the commit hash here.

Thanks for your contribution @zhihuiFan! We appreciate your work and are excited to have this feature in MongoDB.